searches/median_of_medians.py

```diff
@@ -0,0 +1,118 @@
+"""
+A Python implementation of the Median of Medians algorithm
+to select pivots for quick_select, which is efficient for
+calculating the value that would appear in the index of a
+list if it would be sorted, even if it is not already
+sorted. Search in time complexity O(n) at any rank
+deterministically
+https://en.wikipedia.org/wiki/Median_of_medians
+"""
+
+"""
+>>> quick_select([2, 4, 5, 7, 899, 54, 32], 5)
+32
+>>> quick_select([2, 4, 5, 7, 899, 54, 32], 1)
+2
+>>> quick_select([5, 4, 3, 2], 2)
+3
+>>> quick_select([3, 5, 7, 10, 2, 12], 3)
+5
+"""
+
+
+def median_of_five(arr: list) -> int:
+    """
+    >>> median_of_five([2, 4, 5, 7, 899])
+    5
+    >>> median_of_five([5, 7, 899, 54, 32])
+    32
+    >>> median_of_five([5, 4, 3, 2])
+    3
+    >>> median_of_five([3, 5, 7, 10, 2])
+    5
+    """
+    """
+    Return the median of the input list
+    :param arr: Array to find median of
+    :return: median of arr
+    """
+    arr = sorted(arr)
+    return arr[len(arr) // 2]
+
+
+def median_of_medians(arr: list) -> int:
+    """
+    >>> median_of_medians([2, 4, 5, 7, 899, 54, 32])
+    54
+    >>> median_of_medians([5, 7, 899, 54, 32])
+    32
+    >>> median_of_medians([5, 4, 3, 2])
+    4
+    >>> median_of_medians([3, 5, 7, 10, 2, 12])
+    12
+    """
+    """
+    Return a pivot to partition data on by calculating
+    Median of medians of input data
+    :param arr: The data to be checked (a list)
+    :return: median of medians of input array
+    """
```

Comment:
Can you swap the order of the tests and explanation, and condense into a single docstring